data/bookkeeping/block.go

```diff
@@ -624,19 +665,18 @@ func MakeBlock(prev BlockHeader) Block {
 		}
 	}
 
-	bonus := NextBonus(prev, &params)
-
 	// the merkle root of TXs will update when fillpayset is called
 	blk := Block{
 		BlockHeader: BlockHeader{
-			Round:        prev.Round + 1,
-			Branch:       prev.Hash(),
-			TimeStamp:    timestamp,
-			GenesisID:    prev.GenesisID,
-			GenesisHash:  prev.GenesisHash,
-			UpgradeVote:  upgradeVote,
-			UpgradeState: upgradeState,
-			Bonus:        bonus,
+			Round:         prev.Round + 1,
+			Branch:        prev.Hash(),
+			TimeStamp:     timestamp,
+			GenesisID:     prev.GenesisID,
+			GenesisHash:   prev.GenesisHash,
+			UpgradeVote:   upgradeVote,
+			UpgradeState:  upgradeState,
+			Bonus:         NextBonus(prev, &params),
+			CongestionTax: NextCongestionTax(prev.Load, prev.CongestionTax),
```

Comment:
how would it be initially set? like a switch from no congestion (both prev values are zeros) to congested? `NextCongestionTax` uses multiplication so if both inputs is zero it returns zero

Reply:
`Load` will almost always be non-zero.  It is the fraction of a full block.  When you have a Load of 75% (represented by 750,000) `NextCongestionTax` will rise from 0 to 50,000 because we want to go from multiplying the minfee by 1.00 to 1.05.